bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/internal/serializer/CustomReadingInstanceDeserializer.java

```diff
@@ -0,0 +1,107 @@
+    private static final String PROVIDED_FRACTION_DIGITS_SECOND_TARIFF = "provided_fraction_digits_second_tariff";
+
+    @Override
+    public ReadingInstance deserialize(final JsonElement json, final Type typeOfT,
+            final JsonDeserializationContext context) throws JsonParseException {
+        final JsonObject jsonObject = json.getAsJsonObject();
+
+        if (!jsonObject.has(COUNT)) {
+            throw new JsonParseException("Invalid Json Response");
+        }
+
+        ReadingInstance result = new ReadingInstance();
+        int resultCount = Integer.parseInt(jsonObject.get(COUNT).toString());
+
+        // No readings provided yet
+        if (resultCount < 1) {
+            result.setReadingDate(ZonedDateTime.from(Instant.EPOCH));
+            result.setValue(0);
+        }
+
+        // Fist result is the last reading instance
+        JsonObject latestReading = jsonObject.getAsJsonArray(RESULTS).get(0).getAsJsonObject();
+
+        result.setUrl(getStringFromJson(latestReading, URL));
+        result.setAppliedMethod(getStringFromJson(latestReading, APPLIED_METHOD));
+        result.setReadingDate(ZonedDateTime.parse(getStringFromJson(latestReading, READING_DATE)));
+        result.setValue(Double.parseDouble(getStringFromJson(latestReading, VALUE)));
+
+        // Not all meters provide useful data for second tariff and fraction digits , so zero should be used in case of
+        // a null value.
+        String secondTariffValue = getStringFromJson(latestReading, VALUE_SECOND_TARIFF);
+        result.setValueSecondTariff(
+                checkStringForNullValues(secondTariffValue) ? 0 : Double.parseDouble(secondTariffValue));
+
+        String providedFractionDigits = getStringFromJson(latestReading, PROVIDED_FRACTION_DIGITS);
+        result.setProvidedFractionDigits(
+                checkStringForNullValues(providedFractionDigits) ? 0 : Integer.parseInt(providedFractionDigits));
+
+        String secondprovidedFractionDigits = getStringFromJson(latestReading, PROVIDED_FRACTION_DIGITS_SECOND_TARIFF);
+        result.setProvidedFractionDigitsSecondTariff(checkStringForNullValues(secondprovidedFractionDigits) ? 0
+                : Integer.parseInt(secondprovidedFractionDigits));
+
+        return result;
+    }
+
+    /**
+     * Returns the node value and removes possible added quotation marks, which would lead to parse errors.
+     *
+     * @param data The Json source to get the string from
+     * @param key The key for the wanted Json Node
+     * @return The wanted string without unnecessary quotation marks
+     */
+    private String getStringFromJson(JsonObject data, String key) {
+        return data.get(key).toString().replaceAll("\"", "");
+    }
+
+    /**
+     * @param s the striong to check
+     * @return returns true if null values have been found, false otherwise
+     */
+    private boolean checkStringForNullValues(String s) {
+        return (s.isEmpty() || s.equals(null) || s.equals("null"));
```

Comment:
```suggestion
        return (s==null || s.isEmpty() || s.equals("null"));
```

Reply:
This should be the right way. Checking for null is clearly done with `==`, you just have to make sure it is in the correct order.

Reply:
Maybe i didn't try the combination of `s==null`with the other "empty" conditions.
I tried many combinations so maybe this one was lost. 🤷‍♂